packages/plugins/sentry/server/middlewares/sentry-performance.js

```diff
@@ -0,0 +1,84 @@
+'use strict';
+
+const { stripUrlQueryAndFragment } = require("@sentry/utils");
+
+/**
+ * Programmatic sentry-performance middleware. We do not want to expose it in the plugin
+ * @param {{ strapi: import('@strapi/strapi').Strapi }}
+ */
+module.exports = ({ strapi }) => {
+  const sentryService = strapi.plugin('sentry').service('sentry');
+  sentryService.init();
+  const Sentry = sentryService.getInstance();
+
+  if (!Sentry || !sentryService.hasTracingEnabled()) {
+    // initialization failed or tracing is not enabled
+    return;
+  }
+
+  strapi.server.use(async (ctx, next) => {
+    return new Promise((resolve, reject) => {
+      Sentry.runWithAsyncContext(async () => {
+        const hub = Sentry.getCurrentHub();
+        hub.configureScope((scope) =>
+          scope.addEventProcessor((event) =>
+            Sentry.addRequestDataToEvent(event, ctx.request, {
+              include: {
+                user: false,
+              },
+            })
+          )
+        );
+
+        try {
+          await next();
+        } catch (err) {
+          reject(err);
+        }
+        resolve();
```

Comment:
This is unnecessary if we just remove the surrounding promise.

Reply:
good point, removed this (and will also update the docs accordingly 😅 )